label_studio/io_storages/pachyderm/mount_server.py

```diff
@@ -0,0 +1,109 @@
+"""Functionality for interacting with the pachctl mount-server."""
+from dataclasses import dataclass
+from subprocess import Popen
+from time import sleep
+from typing import Dict, Literal, Optional
+
+from requests import get, put, RequestException
+
+MOUNT_MODE = Literal["r", "rw"]
+MOUNT_SERVER_URL = "http://localhost:9002"
+_mount_process: Optional[Popen] = None
+
+
+@dataclass
+class Mount:
+    """Mount object from the mount-server response."""
+    name: str
+    mode: str
+    state: str
+    status: str
+    mountpoint: str
+
+    @classmethod
+    def from_dict(cls, data: Dict) -> "Mount":
+        return cls(
+            name=data['name'],
+            mode=data['mode'],
+            state=data['state'],
+            status=data['status'],
+            mountpoint=data['mountpoint'],
+        )
+
+
+@dataclass
+class Branch:
+    """Branch object from the mount-server response."""
+    name: str
+    mount: Mount
+
+    @classmethod
+    def from_dict(cls, data: Dict) -> "Branch":
+        return cls(
+            name=data['name'],
+            mount=Mount.from_dict(data['mount']),
+        )
+
+
+@dataclass
+class Repo:
+    """Repo object from the mount-server response."""
+    name: str
+    branches: Dict[str, Branch]
+
+    @classmethod
+    def from_dict(cls, data: Dict) -> "Repo":
+        return cls(
+            name=data['name'],
+            branches={
+                name: Branch.from_dict(branch)
+                for name, branch in data['branches'].items()
+            },
+        )
+
+
+def get_repos() -> Dict[str, "Repo"]:
+    """Returns the deserialized response from GET /repos"""
+    response = get(f"{MOUNT_SERVER_URL}/repos")
+    response.raise_for_status()
+    return {
+        name: Repo.from_dict(repo)
+        for name, repo in response.json().items()
+    }
+
+
+def mount_repo(
+    repo: str, branch: str, mode: MOUNT_MODE = 'r', name: Optional[str] = None
+) -> str:
+    """Mount the specified branch of the specified pachyderm repository"""
+    name = name or f"{repo}@{branch}"
+    url = f"{MOUNT_SERVER_URL}/repos/{repo}/{branch}/_mount"
+    put(url, params=dict(name=name, mode=mode)).raise_for_status()
+    return name
+
+
+def unmount_repo(repo: str, branch: str, name: Optional[str] = None) -> None:
+    """Unmount the specified branch of the specified pachyderm repository"""
+    name = name or f"{repo}@{branch}"
+    url = f"{MOUNT_SERVER_URL}/repos/{repo}/{branch}/_unmount"
+    put(url, params=dict(name=name)).raise_for_status()
+
+
+def safe_start_mount_server(*, wait: int = 30) -> None:
+    """Start the mount-server if it is not already started.
+
+    This uses GET /repos to check if the mount-server is started which
+    is the current best option but scales poorly with # of repositories.
```

Comment:
An alternative is to hit the config endpoint (`f"{MOUNT_SERVER_URL}/config"`). If you get a response, mount server is running.

Reply:
I don't think that's api is exposed in pachctl v2.1.4. Do you have any info on whether this was removed from mount or when it will be added to a stable release?

Reply:
Yea you're right. The new endpoints must have been added after 2.1.0 was cut.